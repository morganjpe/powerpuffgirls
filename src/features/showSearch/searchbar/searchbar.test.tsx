import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import SearchBar from './searchbar';
import pretty from 'pretty';

describe('<SearchBar /> component', () => {
  let container: HTMLElement;

  beforeEach(() => {
    const component = render(
      <SearchBar
        isLoading={false}
        search={''}
        handleUserInput={() => {
          console.log('set');
        }}
      />
    );
    container = component.container;
  });

  it('should have a label and input', () => {
    const input = screen.getByLabelText(/search/i);
    expect(input).toHaveAttribute('type', 'text');
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should be accessible', async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
