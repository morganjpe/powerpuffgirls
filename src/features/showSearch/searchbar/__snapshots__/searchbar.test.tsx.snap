// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> component should have a label and input 1`] = `
"<form class=\\"sc-bdnxRM gTmBws\\"><label for=\\"search\\"><span class=\\"hidden\\">search</span>
    <div class=\\"search-container\\"><input id=\\"search\\" name=\\"search\\" placeholder=\\"search...\\" type=\\"text\\" value=\\"\\"></div>
  </label></form>"
`;
